src/cookies.ts

```diff
@@ -0,0 +1,468 @@
+            isChunkLike(name, key),
+          );
+
+          const removeCookieOptions = {
+            ...DEFAULT_COOKIE_OPTIONS,
+            ...options?.cookieOptions,
+            maxAge: 0,
+          };
+
+          if (removeCookies.length > 0) {
+            await setAll(
+              removeCookies.map((name) => ({
+                name,
+                value: "",
+                options: removeCookieOptions,
+              })),
+            );
+          }
+        },
+      },
+    };
+  }
+
+  // This is the server client. It only uses getAll to read the initial
+  // state. Any subsequent changes to the items is persisted in the
+  // setItems and removedItems objects. createServerClient *must* use
+  // getAll, setAll and the values in setItems and removedItems to
+  // persist the changes *at once* when appropriate (usually only when
+  // the TOKEN_REFRESHED, USER_UPDATED or SIGNED_OUT events are fired by
+  // the Supabase Auth client).
+  return {
+    getAll,
+    setAll,
+    setItems,
+    removedItems,
+    storage: {
+      // to signal to the libraries that these cookies are
+      // coming from a server environment and their value
+      // should not be trusted
+      isServer: true,
+      getItem: async (key: string) => {
+        if (typeof setItems[key] === "string") {
+          return setItems[key];
+        }
+
+        if (removedItems[key]) {
+          return null;
+        }
+
+        const allCookies = await getAll([key]);
+        const chunkedCookie = await combineChunks(
+          key,
+          async (chunkName: string) => {
+            const cookie =
+              allCookies?.find(({ name }) => name === chunkName) || null;
+
+            if (!cookie) {
+              return null;
+            }
+
+            return cookie.value;
+          },
+        );
+
+        if (!chunkedCookie) {
+          return null;
+        }
+
+        let decoded = chunkedCookie;
+
+        if (chunkedCookie.startsWith("base64:")) {
+          decoded = stringFromBase64URL(
+            chunkedCookie.substring("base64:".length),
+          );
+        }
+
+        return decoded;
+      },
+      setItem: async (key: string, value: string) => {
+        // We don't have an `onAuthStateChange` event that can let us know that
+        // the PKCE code verifier is being set. Therefore, if we see it being
+        // set, we need to apply the storage (call `setAll` so the cookie is
+        // set properly).
+        if (key.endsWith("-code-verifier")) {
+          await applyServerStorage(
+            {
+              getAll,
+              setAll,
+              // pretend only that the code verifier was set
+              setItems: { [key]: value },
+              // pretend that nothing was removed
+              removedItems: {},
+            },
+            {
+              cookieOptions: options?.cookieOptions ?? null,
+              cookieEncoding,
+            },
+          );
+        }
```

Comment:
This is how the code verifier is persisted when generated on the server. See `removeItem` for the similar case, though in reality that one is rarely to come by on its own.